sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -247,7 +247,28 @@ case class DropTableCommand(
     } else if (ifExists) {
       // no-op
     } else {
-      throw QueryCompilationErrors.tableOrViewNotFoundError(tableName.identifier)
+      throw QueryCompilationErrors.noSuchTableError(
+        tableName.catalog.toSeq ++ tableName.database :+ tableName.table)
+    }
+    Seq.empty[Row]
+  }
+}
+
+case class DropTempViewCommand(ident: Identifier) extends LeafRunnableCommand {
```

Comment:
v1 only, right?

Reply:
temp view is a Spark internal thing and is unrelated to data source, so it's neither v1 nor v2.

Reply:
Oh, I see. I read the comment "v1 DROP TABLE supports temp view." wrongly. There is also other pattern statement for v2 going to `DropTempViewCommand`.